### PR TITLE
Feature 113/questions create - closes issue 113

### PR DESCRIPTION
**Pull request by hwilliams314**

### CODE REVIEW

File: server/src/main/java/com/objectcomputing/checkins/services/questions/QuestionRepository.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.objectcomputing.checkins.services.questions;
+
+import io.micronaut.data.jdbc.annotation.JdbcRepository;
+import io.micronaut.data.model.query.builder.sql.Dialect;
+import io.micronaut.data.repository.CrudRepository;
+import jdk.internal.jline.internal.Nullable;
```
Comment (mkimberlin) [inline server/src/main/java/com/objectcomputing/checkins/services/questions/QuestionRepository.java:6]: import javax.annotation.Nullable;

Comment (hwilliams314) [reply server/src/main/java/com/objectcomputing/checkins/services/questions/QuestionRepository.java]: done

File: server/src/main/java/com/objectcomputing/checkins/services/questions/Question.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+import javax.validation.constraints.NotNull;
+import java.util.Objects;
+import java.util.UUID;
+
+
+@Entity
+@Table(name ="questions")
+public class Question {
+
+    public Question(@NotBlank @NotNull String text) {
```
Comment (chrismckiernan) [inline server/src/main/java/com/objectcomputing/checkins/services/questions/Question.java:20]: Only @NotBlank is needed.
https://javaee.github.io/javaee-spec/javadocs/javax/validation/constraints/NotBlank.html

File: server/src/main/java/com/objectcomputing/checkins/services/questions/QuestionController.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import java.util.List;
+import java.util.UUID;
+
+@Controller("/questions")
+@Secured(SecurityRule.IS_ANONYMOUS)
+@Produces(MediaType.APPLICATION_JSON)
+@Tag(name="questions")
+public class QuestionController {
+
+    //    an endpoint is created for create
```
Comment (chrismckiernan) [inline server/src/main/java/com/objectcomputing/checkins/services/questions/QuestionController.java:24]: Placeholder comments?

File: server/src/main/java/com/objectcomputing/checkins/services/questions/QuestionController.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        if (newQuestion == null) {
+            return HttpResponse.status(HttpStatus.valueOf(409), "already exists");
+        } else {
+            return HttpResponse
+                    .created(newQuestion)
+                    .headers(headers -> headers.location(location(newQuestion.getQuestionid())));
+        }
+    }
+    //  I know these are commented out - I added them before splitting the story and will
+    // need them in the future - please ignore for now
```
Comment (chrismckiernan) [inline server/src/main/java/com/objectcomputing/checkins/services/questions/QuestionController.java:55]: Ignoring for now =)

File: server/src/main/java/com/objectcomputing/checkins/services/questions/QuestionServices.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+import javax.inject.Singleton;
+import java.util.List;
+import java.util.UUID;
+
+@Singleton
+public class QuestionServices {
+
+    private static final Logger LOG = LoggerFactory.getLogger(QuestionServices.class);
+
+    //    an endpoint is created for read all
```
Comment (chrismckiernan) [inline server/src/main/java/com/objectcomputing/checkins/services/questions/QuestionServices.java:16]: These comments needed?

File: server/src/main/java/com/objectcomputing/checkins/services/questions/Question.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    @Id
+    @Column(name="questionid")
+    @AutoPopulated
+    @TypeDef(type=DataType.STRING)
+    private UUID questionid;
+
+    @NotBlank
+    @NotNull
```
Comment (chrismckiernan) [inline server/src/main/java/com/objectcomputing/checkins/services/questions/Question.java:35]: only not blank needed for above reasons.

File: server/src/main/java/com/objectcomputing/checkins/services/questions/Question.java (hunk 2 above)
Comment (hwilliams314) [reply server/src/main/java/com/objectcomputing/checkins/services/questions/Question.java:20]: done

File: server/src/main/java/com/objectcomputing/checkins/services/questions/QuestionController.java (hunk 3 above)
Comment (hwilliams314) [reply server/src/main/java/com/objectcomputing/checkins/services/questions/QuestionController.java:24]: yes, easier than flipping back and forth to the issue. They will be removed before final pr, I promise.

File: server/src/main/java/com/objectcomputing/checkins/services/questions/QuestionController.java (hunk 4 above)
Comment (hwilliams314) [reply server/src/main/java/com/objectcomputing/checkins/services/questions/QuestionController.java:55]: you are so kind

File: server/src/main/java/com/objectcomputing/checkins/services/questions/QuestionServices.java (hunk 5 above)
Comment (hwilliams314) [reply server/src/main/java/com/objectcomputing/checkins/services/questions/QuestionServices.java:16]: I guess not. They have been removed.

File: server/src/test/java/com/objectcomputing/checkins/questions/QuestionServicesTest.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+    @BeforeEach
+    void setup() {
+        itemUnderTest.setQuestionRepository(mockQuestionRepository);
+        reset(mockQuestionRepository);
+        reset(mockQuestion);
+    }
+
+    @Test
+    public void testSaveSkill() {
```
Comment (chrismckiernan) [inline server/src/test/java/com/objectcomputing/checkins/questions/QuestionServicesTest.java:55]: testSaveQuestion?

File: server/src/main/java/com/objectcomputing/checkins/services/questions/Question.java (hunk 6 above)
Comment (hwilliams314) [reply server/src/main/java/com/objectcomputing/checkins/services/questions/Question.java:35]: done

File: server/src/test/java/com/objectcomputing/checkins/questions/QuestionServicesTest.java (hunk 7 above)
Comment (hwilliams314) [reply server/src/test/java/com/objectcomputing/checkins/questions/QuestionServicesTest.java:55]: Oops! fixed

**Pull request merged**